### PR TITLE
[TACHYON-934] Made TachyonFileSystem methods throw better exceptions

### PR DESCRIPTION
**Pull request by manugoyal**
https://tachyon.atlassian.net/browse/TACHYON-934

We were turning everything into an IOException, which can make it hard
for callers to determine what went wrong in an operation. So now the
TachyonFileSystem and related classes don't wrap exceptions in an
IOException.

Also the FileSystemMasterClient uses an exponential backoff retry,
rather than looping forever, before throwing an exception and exiting.

There are a lot of miscellaneous refactors for deprecated code and tests
to get this to compile.

### CODE REVIEW

**Comment by apc999** [comment]
Thanks @manugoyal 

For change of this size, could you link its JIRA (if it is not there yet, could you create one) here?

Thanks

**Comment by manugoyal** [comment]
Created a JIRA: https://tachyon.atlassian.net/browse/TACHYON-934

**Comment by calvinjia** [comment]
@manugoyal Could you reopen this PR to master please?

File: clients/unshaded/src/main/java/tachyon/client/TachyonFS.java [hunk 1]
```diff
@@ -307,7 +312,7 @@ public synchronized int createDependency(List<String> parents, List<String> chil
       String commandPrefix, List<ByteBuffer> data, String comment, String framework,
       String frameworkVersion, int dependencyType, long childrenBlockSizeByte) throws IOException {
     return mFSMasterClient.user_createDependency(parents, children, commandPrefix, data, comment,
-        framework, frameworkVersion, dependencyType, childrenBlockSizeByte);
+            framework, frameworkVersion, dependencyType, childrenBlockSizeByte);
```
Comment (calvinjia) [inline clients/unshaded/src/main/java/tachyon/client/TachyonFS.java]: indentation is off here.

File: shell/src/test/java/tachyon/shell/TFsShellUtilsTest.java [hunk 2]
```diff
@@ -140,12 +145,12 @@ public static String resetLocalFileHierarchy(LocalTachyonCluster localTachyonClu
...
-     *                                        └── foobar2
-     *                                ├── bar |
-     *                                        └── foobar3
-     *                                └── foobar4
+                ├── foo |
+                        ├── foobar1
+                        └── foobar2
+                ├── bar |
+                        └── foobar3
+                └── foobar4
```
Comment (calvinjia) [inline shell/src/test/java/tachyon/shell/TFsShellUtilsTest.java]: Could you revert the formatting change (I think the lines are supposed to be coming down from `/testWildCards`?

**Pull request closed without merging**